connectd/connect.c

```diff
@@ -1073,6 +1073,7 @@ static void try_reach_one_addr(struct reaching *reach)
 {
  	int fd, af;
 	bool use_proxy = reach->daemon->use_proxy_always;
+	const struct wireaddr_internal *addr = &reach->addrs[reach->addrnum];
```

Comment:
This assignment needs to move down a bit, otherwise the `reach->addrnum == tal_count(reach->addrs)` blows up in our face.